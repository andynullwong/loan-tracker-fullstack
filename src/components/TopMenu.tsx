"use client";

import React from "react";
import {
  BankOutlined,
  CopyOutlined,
  FormOutlined,
  HomeOutlined,
  UserAddOutlined,
  UserOutlined,
  UsergroupAddOutlined,
} from "@ant-design/icons";
import type { MenuProps } from "antd";
import { Alert, Menu } from "antd";
import { useUserContext } from "@/context/user.context";
import Link from "next/link";

const TopMenu = () => {
  const { user, setUser } = useUserContext();
  const items: MenuProps["items"] = [
    {
      label: <Link href="/">Home</Link>,
      key: "home",
      icon: <HomeOutlined />,
    },
    {
      label: "Users",
      key: "users",
      icon: <UserOutlined />,
      children: [
        {
          label: <Link href="/users">Get All Users</Link>,
          key: "users-get-all",
          icon: <UsergroupAddOutlined />,
        },
        {
          label: <Link href="/users/create">Create User</Link>,
          key: "users-create",
          icon: <UserAddOutlined />,
        },
      ],
    },
    {
      label: "Loans",
      key: "loans",
      icon: <BankOutlined />,
      children: [
        {
          label: <Link href={`/users/${user}`}>My Loans</Link>,
          key: "loans-get-all",
          icon: <CopyOutlined />,
        },
        {
          label: <Link href="/loans/create">Create Loan</Link>,
          key: "loans-create",
          icon: <FormOutlined />,
        },
      ],
    },
  ];

  return (
    <>
      <Menu mode="horizontal" items={items} />
      <Alert message={`Logged in as User ${user}`} type="info" />
    </>
  );
};

export default TopMenu;
