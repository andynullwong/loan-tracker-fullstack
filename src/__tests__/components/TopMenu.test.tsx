import { act, render } from "@testing-library/react";
import TopMenu from "../../components/TopMenu";
import "@testing-library/jest-dom";

test("renders TopMenu correctly", () => {
  let component = null;

  act(() => {
    component = render(<TopMenu />);
  });
  expect(component).toMatchSnapshot();
});
