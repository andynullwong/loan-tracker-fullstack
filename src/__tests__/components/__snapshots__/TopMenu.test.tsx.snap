// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TopMenu correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-light css-dev-only-do-not-override-1vr7spz"
        data-menu-list="true"
        role="menu"
        tabindex="0"
      >
        <li
          class="ant-menu-overflow-item ant-menu-item"
          data-menu-id="rc-menu-uuid-test-home"
          role="menuitem"
          style="opacity: 1; order: 0;"
          tabindex="-1"
        >
          <span
            aria-label="home"
            class="anticon anticon-home ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="home"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            <a
              href="/"
            >
              Home
            </a>
          </span>
        </li>
        <li
          class="ant-menu-overflow-item ant-menu-submenu ant-menu-submenu-horizontal"
          role="none"
          style="opacity: 1; order: 1;"
        >
          <div
            aria-controls="rc-menu-uuid-test-users-popup"
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="menuitem"
            tabindex="-1"
          >
            <span
              aria-label="user"
              class="anticon anticon-user ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="user"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                />
              </svg>
            </span>
            <span
              class="ant-menu-title-content"
            >
              Users
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
        <li
          class="ant-menu-overflow-item ant-menu-submenu ant-menu-submenu-horizontal"
          role="none"
          style="opacity: 1; order: 2;"
        >
          <div
            aria-controls="rc-menu-uuid-test-loans-popup"
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            role="menuitem"
            tabindex="-1"
          >
            <span
              aria-label="bank"
              class="anticon anticon-bank ant-menu-item-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="bank"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M894 462c30.9 0 43.8-39.7 18.7-58L530.8 126.2a31.81 31.81 0 00-37.6 0L111.3 404c-25.1 18.2-12.2 58 18.8 58H192v374h-72c-4.4 0-8 3.6-8 8v52c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-52c0-4.4-3.6-8-8-8h-72V462h62zM512 196.7l271.1 197.2H240.9L512 196.7zM264 462h117v374H264V462zm189 0h117v374H453V462zm307 374H642V462h118v374z"
                />
              </svg>
            </span>
            <span
              class="ant-menu-title-content"
            >
              Loans
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
        <li
          aria-hidden="true"
          class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
          role="none"
          style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
        >
          <div
            aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            data-menu-id="rc-menu-uuid-test-rc-menu-more"
            role="menuitem"
            tabindex="-1"
          >
            <span
              aria-label="ellipsis"
              class="anticon anticon-ellipsis"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="ellipsis"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                />
              </svg>
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
      </ul>
      <div
        aria-hidden="true"
        style="display: none;"
      />
      <div
        class="ant-alert ant-alert-info ant-alert-no-icon css-dev-only-do-not-override-1vr7spz"
        data-show="true"
        role="alert"
      >
        <div
          class="ant-alert-content"
        >
          <div
            class="ant-alert-message"
          >
            Logged in as User undefined
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <ul
      class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-light css-dev-only-do-not-override-1vr7spz"
      data-menu-list="true"
      role="menu"
      tabindex="0"
    >
      <li
        class="ant-menu-overflow-item ant-menu-item"
        data-menu-id="rc-menu-uuid-test-home"
        role="menuitem"
        style="opacity: 1; order: 0;"
        tabindex="-1"
      >
        <span
          aria-label="home"
          class="anticon anticon-home ant-menu-item-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="home"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
            />
          </svg>
        </span>
        <span
          class="ant-menu-title-content"
        >
          <a
            href="/"
          >
            Home
          </a>
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-submenu ant-menu-submenu-horizontal"
        role="none"
        style="opacity: 1; order: 1;"
      >
        <div
          aria-controls="rc-menu-uuid-test-users-popup"
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="menuitem"
          tabindex="-1"
        >
          <span
            aria-label="user"
            class="anticon anticon-user ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="user"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            Users
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-submenu ant-menu-submenu-horizontal"
        role="none"
        style="opacity: 1; order: 2;"
      >
        <div
          aria-controls="rc-menu-uuid-test-loans-popup"
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          role="menuitem"
          tabindex="-1"
        >
          <span
            aria-label="bank"
            class="anticon anticon-bank ant-menu-item-icon"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="bank"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M894 462c30.9 0 43.8-39.7 18.7-58L530.8 126.2a31.81 31.81 0 00-37.6 0L111.3 404c-25.1 18.2-12.2 58 18.8 58H192v374h-72c-4.4 0-8 3.6-8 8v52c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-52c0-4.4-3.6-8-8-8h-72V462h62zM512 196.7l271.1 197.2H240.9L512 196.7zM264 462h117v374H264V462zm189 0h117v374H453V462zm307 374H642V462h118v374z"
              />
            </svg>
          </span>
          <span
            class="ant-menu-title-content"
          >
            Loans
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
      <li
        aria-hidden="true"
        class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
        role="none"
        style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
      >
        <div
          aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          data-menu-id="rc-menu-uuid-test-rc-menu-more"
          role="menuitem"
          tabindex="-1"
        >
          <span
            aria-label="ellipsis"
            class="anticon anticon-ellipsis"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="ellipsis"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
              />
            </svg>
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
    </ul>
    <div
      aria-hidden="true"
      style="display: none;"
    />
    <div
      class="ant-alert ant-alert-info ant-alert-no-icon css-dev-only-do-not-override-1vr7spz"
      data-show="true"
      role="alert"
    >
      <div
        class="ant-alert-content"
      >
        <div
          class="ant-alert-message"
        >
          Logged in as User undefined
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
